synapse/models/dns.py

```diff
@@ -101,211 +107,76 @@ def getModelDefs(self):
                 )),
-                ('inet:dns:req', {}, (
-                    ('client', ('inet:client', {}), {
-                        'ro': True,
-                        'doc': 'The inet:addr which requested the FQDN',
-                    }),
-                    ('client:ipv4', ('inet:ipv4', {}), {
-                        'ro': True,
-                        'doc': 'The IPv4 of the client.',
-                    }),
-                    ('client:ipv6', ('inet:ipv6', {}), {
-                        'ro': True,
-                        'doc': 'The IPv6 of the client.',
-                    }),
-                    ('fqdn', ('inet:fqdn', {}), {
-                        'ro': True,
-                        'doc': 'The requested FQDN',
-                    }),
-                    ('type', ('inet:dns:type', {}), {
-                        'ro': True,
-                        'doc': 'The type of DNS record requested.',
-                    }),
 
+                ('inet:dns:query', {}, (
+                    ('client', ('inet:client', {}), {}),
+                    ('name', ('inet:dns:name', {}), {}),
+                    ('type', ('int', {}), {}),
                 )),
-                ('inet:dns:look', {}, (
-                    ('time', ('time', ()), {
-                        'ro': True,
-                        'doc': 'The time thel ookup occured.',
-                    }),
-                    ('client', ('inet:client', {}), {
-                        'doc': 'The client that requested the lookup.',
-                    }),
-                    ('client:ipv4', ('inet:ipv4', {}), {
-                        'doc': 'The IPv4 of the client that requested the lookup.',
-                    }),
-                    ('client:ipv6', ('inet:ipv6', {}), {
-                        'doc': 'The IPv6 of the client that requested the lookup.',
-                    }),
-                    ('client:port', ('inet:ipv4', {}), {
-                        'doc': 'The port of the client that requested the lookup.',
-                    }),
-                    ('server', ('inet:server', {}), {
-                        'doc': 'The server that responded to the lookup.',
-                    }),
-                    ('server:ipv4', ('inet:ipv4', {}), {
-                        'doc': 'The IPv4 of the server that responded to the lookup.',
-                    }),
-                    ('server:ipv6', ('inet:ipv6', {}), {
-                        'doc': 'The IPv6 of the server that responded to the lookup.',
-                    }),
-                    ('server:port', ('inet:ipv4', {}), {
-                        'doc': 'The port of the server that responded to the lookup.',
-                    }),
+
+                ('inet:dns:request', {}, (
+
+                    ('time', ('time', {}), {}),
+                    ('query', ('inet:dns:query', {}), {}),
+
+                    ('server', ('inet:server', {}), {}),
+
                     ('rcode', ('int', {}), {
```

Comment:
rcode ("response code") should not be part of an inet:dns:REQUEST. Looks like query type ("type") is already part of the 'inet:dns:query' secondary prop.